[Bug]: map_batches_test.py fails on 32bit archies

### Describe the bug

I am [packing py3-narwhals package](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/100008) for Alpine Linux.

The `x86`, `armhf`, and `armv7` targets all fail on same test:

```
FAILED tests/expr_and_series/map_batches_test.py::test_map_batches_exception[pandas] - AssertionError: Regex pattern did not match.
  Expected regex: "`map(?:_batches)?` with `returns_scalar=False` must return a Series; found 'numpy.int64'.\n\nIf `returns_scalar` is set to `True`, a returned value can be a scalar value."
  Actual message: "`map_batches` with `returns_scalar=False` must return a Series; found 'numpy.int32'.\n\nIf `returns_scalar` is set to `True`, a returned value can be a scalar value."
``` 

### Steps or code to reproduce the bug

Run tests

### Expected results

No error is thrown

### Actual results

FAILED tests/expr_and_series/map_batches_test.py::test_map_batches_exception[pandas] - AssertionError: Regex pattern did not match.
  Expected regex: "`map(?:_batches)?` with `returns_scalar=False` must return a Series; found 'numpy.int64'.\n\nIf `returns_scalar` is set to `True`, a returned value can be a scalar value."
  Actual message: "`map_batches` with `returns_scalar=False` must return a Series; found 'numpy.int32'.\n\nIf `returns_scalar` is set to `True`, a returned value can be a scalar value."

### Please run narwhals.show_versions() and enter the output below.

```shell
2.19.0
```

### Relevant log output

```shell
=================================== FAILURES ===================================
______________________ test_map_batches_exception[pandas] ______________________
constructor_eager = <function pandas_constructor at 0xe6759fa0>
request = <FixtureRequest for <Function test_map_batches_exception[pandas]>>
    def test_map_batches_exception(
        constructor_eager: ConstructorEager, request: pytest.FixtureRequest
    ) -> None:
        if "polars" in str(constructor_eager) and POLARS_VERSION < (1, 32, 3):
            request.applymarker(pytest.mark.xfail)
    
        df = nw.from_native(constructor_eager(data))
        msg = (
            r"`map(?:_batches)?` with `returns_scalar=False` must return a Series; found "
            "'numpy.int64'.\n\nIf `returns_scalar` is set to `True`, a returned value can be "
            "a scalar value."
        )
    
        with pytest.raises(TypeError, match=msg):
>           df.select(nw.all().map_batches(lambda s: s.to_numpy().argmax()))
tests/expr_and_series/map_batches_test.py:90: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
narwhals/dataframe.py:1530: in select
    return super().select(*exprs, **named_exprs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
narwhals/dataframe.py:240: in select
    return self._with_compliant(self._compliant_frame.select(*compliant_exprs))
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
narwhals/_pandas_like/dataframe.py:448: in select
    new_series = self._evaluate_exprs(*exprs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
narwhals/_compliant/dataframe.py:367: in _evaluate_exprs
    return tuple(chain.from_iterable(self._evaluate_expr(expr) for expr in exprs))  # pyright: ignore[reportArgumentType]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
narwhals/_compliant/dataframe.py:367: in <genexpr>
    return tuple(chain.from_iterable(self._evaluate_expr(expr) for expr in exprs))  # pyright: ignore[reportArgumentType]
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^
narwhals/_compliant/dataframe.py:380: in _evaluate_expr
    result = expr(self)
             ^^^^^^^^^^
narwhals/_compliant/expr.py:247: in __call__
    return self._call(df)
           ^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
df = <narwhals._pandas_like.dataframe.PandasLikeDataFrame object at 0xe3a48c90>
    def func(df: EagerDataFrameT) -> Sequence[EagerSeriesT]:
        udf_series_in = self(df)
        output_names = (input_series.name for input_series in udf_series_in)
        udf_series_out = tuple(function(series) for series in udf_series_in)
        _first_in, _first_out = udf_series_in[0], udf_series_out[0]
    
        result: Sequence[EagerSeriesT]
        it = zip_strict(udf_series_out, output_names)
        if is_numpy_array(_first_out) or is_numpy_scalar(_first_out):
            from_numpy = partial(_first_in.from_numpy, context=self)
            result = tuple(from_numpy(arr).alias(out_name) for arr, out_name in it)
        elif isinstance(_first_out, _first_in.__class__):  # compliant series
            result = tuple(series.alias(out_name) for series, out_name in it)
        else:  # If everything else fails, assume scalar case
            from_scalar = _first_in._from_scalar
            result = tuple(from_scalar(val).alias(out_name) for val, out_name in it)
    
        if return_dtype is not None:
            result = tuple(series.cast(return_dtype) for series in result)
    
        is_scalar_result = tuple(len(r) == 1 for r in result)
        if (not returns_scalar) and any(is_scalar_result) and (len(df) > 1):
            _idx = is_scalar_result.index(True)  # Index of first result with length 1
            _type = type(udf_series_out[_idx])
            msg = (
                "`map_batches` with `returns_scalar=False` must return a Series; "
                f"found '{qualified_type_name(_type)}'.\n\nIf `returns_scalar` "
                "is set to `True`, a returned value can be a scalar value."
            )
>           raise TypeError(msg)
E           TypeError: `map_batches` with `returns_scalar=False` must return a Series; found 'numpy.int32'.
E           
E           If `returns_scalar` is set to `True`, a returned value can be a scalar value.
narwhals/_compliant/expr.py:775: TypeError
During handling of the above exception, another exception occurred:
constructor_eager = <function pandas_constructor at 0xe6759fa0>
request = <FixtureRequest for <Function test_map_batches_exception[pandas]>>
    def test_map_batches_exception(
        constructor_eager: ConstructorEager, request: pytest.FixtureRequest
    ) -> None:
        if "polars" in str(constructor_eager) and POLARS_VERSION < (1, 32, 3):
            request.applymarker(pytest.mark.xfail)
    
        df = nw.from_native(constructor_eager(data))
        msg = (
            r"`map(?:_batches)?` with `returns_scalar=False` must return a Series; found "
            "'numpy.int64'.\n\nIf `returns_scalar` is set to `True`, a returned value can be "
            "a scalar value."
        )
    
>       with pytest.raises(TypeError, match=msg):
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       AssertionError: Regex pattern did not match.
E         Expected regex: "`map(?:_batches)?` with `returns_scalar=False` must return a Series; found 'numpy.int64'.\n\nIf `returns_scalar` is set to `True`, a returned value can be a scalar value."
E         Actual message: "`map_batches` with `returns_scalar=False` must return a Series; found 'numpy.int32'.\n\nIf `returns_scalar` is set to `True`, a returned value can be a scalar value."
tests/expr_and_series/map_batches_test.py:89: AssertionError
```

--- a/tests/expr_and_series/map_batches_test.py
+++ b/tests/expr_and_series/map_batches_test.py
@@ -82,7 +82,8 @@ def test_map_batches_exception(
     df = nw.from_native(constructor_eager(data))
     msg = (
         r"`map(?:_batches)?` with `returns_scalar=False` must return a Series; found "
-        "'numpy.int64'.\n\nIf `returns_scalar` is set to `True`, a returned value can be "
+        r"'numpy\.int(?:32|64)'\."
+        "\n\nIf `returns_scalar` is set to `True`, a returned value can be "
         "a scalar value."
     )
 

